src/blockchain/different-networks/tron-strategy.service.ts

```diff
@@ -0,0 +1,136 @@
+import { PrivateServerSignTransactionDto, SignTransactionDto } from 'rox-custody_common-modules/libs/interfaces/sign-transaction.interface';
+import {
+  IBlockChainPrivateServer,
+  InitBlockChainPrivateServerStrategies,
+  IWalletKeys,
+} from '../interfaces/blockchain.interface';
+
+import { TronWeb } from 'tronweb';
+import { Chain } from 'viem';
+import { getChainFromNetwork } from 'rox-custody_common-modules/blockchain/global-commons/get-network-chain';
+import { supportedNetworks } from 'rox-custody_common-modules/blockchain/global-commons/supported-networks.enum';
+import { TransientService } from 'utils/decorators/transient.decorator';
+import { forwardRef, Inject } from '@nestjs/common';
+import { KeysManagerService } from 'src/keys-manager/keys-manager.service';
+import { NonceManagerService } from 'src/keys-manager/nonce-manager.service';
+import { CustodySignedTransaction } from 'rox-custody_common-modules/libs/interfaces/custom-signed-transaction.type';
+import { AssetType, CommonAsset } from 'rox-custody_common-modules/libs/entities/asset.entity';
+import { CommonNetwork } from 'rox-custody_common-modules/libs/entities/network.entity';
+
+const tronMainnet = 'https://api.trongrid.io';
+const tronShastaTestnet = 'https://api.shasta.trongrid.io';
+const tronNileTestnet = 'https://nile.trongrid.io';
+
+@TransientService()
+export class TronStrategyService implements IBlockChainPrivateServer {
+  private tronWeb;
+  private host: string;
+  private asset: CommonAsset;
+  private network: CommonNetwork;
+  private chain: Chain;
+  private gasStationPk: string;
+
+  constructor(
+    @Inject(forwardRef(() => KeysManagerService))
+    private readonly keyManagerService: KeysManagerService,
+    private readonly nonceManager: NonceManagerService,
+  ) {}
+
+  async init(initData: InitBlockChainPrivateServerStrategies): Promise<void> {
+    const { asset, network, privateKey } = initData;
+    this.network = network;
+    this.asset = asset;
+    const networkObject = getChainFromNetwork(network.networkId);
+
+    this.chain = networkObject.chain;
+
+    switch (network.networkId) {
+      case supportedNetworks.tron:
+        this.host = tronMainnet;
+        break;
+      case supportedNetworks.shastaTestnet:
+        this.host = tronShastaTestnet;
+        break;
+      case supportedNetworks.nileTestnet:
+        this.host = tronNileTestnet;
+        break;
+    }
+
+    this.tronWeb = new TronWeb({
+      fullHost: this.host,
+      privateKey: privateKey,
```

Comment:
why do you need the private key here it's not resonable to have the private key while init since this can be used for generate wallet operation for example